### PR TITLE
[WIP] Adding sample weight to Multi-label Confusion matrix

### PR DESCRIPTION
**Pull request by SSaishruthi**
This PR related to #396

### CODE REVIEW

**Comment by SSaishruthi** [comment]
@Squadrick 
Adding the same comment here as well just for reference.

I have created this PR to add sample weight parameter to F1. Right now, I did not modify the code to take sample weight into account. Just added the param so that it can be used when model training as per the issue. 

I am referring https://github.com/tensorflow/tensorflow/blob/7dcd936389a16028843c822315a8e3a0f7591653/tensorflow/python/keras/metrics.py#L113 for adding broadcasting weights. 

Would like to know your comments on further steps and if the above approach is ok to try.

Thanks in advance.

**Comment by Squadrick** [comment]
Could you add a `TODO` so that we can get to that later? Also, having a test to check whether each `Metric` works with the Keras API would be very beneficial.

**Comment by SSaishruthi** [comment]
Thanks will update the PR

**Comment by Squadrick** [comment]
@SSaishruthi Yup. We need to check everything with the Keras API.

**Comment by SSaishruthi** [comment]
@Squadrick I have the updated code ready but do you think the confusion matrix will be used as an evaluation metric when training (fit)? 
Usually, they are either used in the validation or test step.

**Comment by SSaishruthi** [comment]
@Squadrick I have updated the code. But did not add keras test as I wanted to confirm my above question

**Comment by SSaishruthi** [comment]
@Squadrick @seanpmorgan any update on this?

**Comment by SSaishruthi** [comment]
@seanpmorgan Any update on my above question?

**Comment by Squadrick** [comment]
@SSaishruthi Sorry about the delay. You're right, a confusion matrix is usually used during validation, and much less during training. However, if it's not too much extra work, we'd like to support both scenarios in case someone wants to use it during training.

**Comment by gabrieldemarmiesse** [comment]
I'll close this pull request as it has been inactive for a while now. Feel free to comment if you want me to reopen the pull request.
